src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -214,40 +218,55 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
 
         #region NuGetProject
 
-        public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        public async override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
-            PackageReference[] installedPackages;
-
             var packageSpec = GetPackageSpec();
-            if (packageSpec != null)
-            {
-                installedPackages = GetPackageReferences(packageSpec);
-            }
-            else
-            {
-                installedPackages = new PackageReference[0];
-            }
-
-            return Task.FromResult<IEnumerable<PackageReference>>(installedPackages);
+            return packageSpec != null ? await GetPackageReferences(packageSpec) : new PackageReference[0];
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private async Task<IEnumerable<PackageReference>> GetPackageReferences(PackageSpec packageSpec)
         {
             var frameworkSorter = new NuGetFrameworkSorter();
 
+            var assetsFilePath = await GetAssetsFilePathAsync();
```

Comment:
This NetCorePackageReferenceProject change comes from Martin's [PR](https://github.com/NuGet/NuGet.Client/pull/3539). He already had PR for reference type unit tests and I'm using some of his code only for reference type project unit test.